windows/src/engine/keyman32/k32_lowlevelkeyboardhook.cpp

```diff
@@ -145,7 +155,18 @@ LRESULT _kmnLowLevelKeyboardProc(
   SendDebugMessageFormat(0, sdmAIDefault, 0, "kmnLowLevelKeyboardProc: wparam: %x  lparam: %x [vk:%s scan:%x flags:%x extra:%x]", wParam, lParam, Debug_VirtualKey((WORD) hs->vkCode), hs->scanCode, hs->flags, hs->dwExtraInfo);   // I4674
 
   DWORD Flag = 0;
-  if (UseRegisterHotkey()) {
+  if (!UseCachedHotkeyModifierState()) {
+    // #5190: Don't cache modifier state because sometimes we won't receive
+    // modifier change events (e.g. on lock screen)
+    FHotkeyShiftState = 0;
+    if (GetKeyState(VK_LCONTROL) < 0) FHotkeyShiftState |= HK_CTRL;
```

Comment:
ll. 184-185 says 
> The right modifiers are not used for hotkeys...

Does that mean VK_RCONTROL, VK_RMENU, and VK_RSHIFT can be ignored?

Reply:
No, because if the user is pressing them then we don't want to activate the hotkey.